app/javascript/components/sidebar/sectors/component.js

```diff
@@ -12,39 +13,31 @@ class SectorsComponent extends React.Component {
   static propTypes = {
     fetchSectors: PropTypes.func.isRequired,
     setSelectedSector: PropTypes.func.isRequired,
-    setSelectedLayer: PropTypes.func.isRequired,
+    setSelectedLayers: PropTypes.func.isRequired,
     sectorTitles: PropTypes.array.isRequired,
     list: PropTypes.array.isRequired,
     selectedLayers: PropTypes.object.isRequired
   }
 
-  componentWillMount() {
-    this.props.fetchSectors();
-  }
-
   clickSector(sector) {
     this.props.setSelectedSector(sector);
   }
 
-  handleSelectedType(sector, type) {
-    const types = this.props.selectedLayers;
-    // types[sector] = types[sector] ? types[sector].includes(type) ? types[sector].splice(types[sector].indexOf(type), 1) : types[sector].concat([type]) : [type];
+  handleSelectedType(sectorLayer) {
+    let layers = this.props.selectedLayers;
+    const id = sectorLayer.type_id;
 
-    if (types[sector]) {
-      if (types[sector].includes(type)) {
-        types[sector].splice(types[sector].indexOf(type), 1);
-      } else {
-        types[sector] = types[sector].concat([type]);
-      }
+    if (layers.includes(id)) {
+      layers.splice(layers.indexOf(id), 1);
     } else {
-      types[sector] = [type];
+      layers = layers.concat([id]);
```

Comment:
This is ok, you can also use `layers.push(id)`. So you won't need to use `let layers`